### DISCUSSION TITLE
Weird issue with variable in Actions

### DISCUSSION THREAD

**Discussion by markyarnold** [community/community#38211] [Other Feature Feedback, Questions, & Ideas]
Hi all; hope you're well.

I have an actions workflow which exports 21 variables, all acquired from secrets.

The workflow has been working fine, until I added this:

> export REACT_APP_COG_AUTHORISATION=${{ secrets.V5UAT_REACT_APP_COG_AUTH }}

The secret is defined fine.

The workflow fails with the following:

/__w/_temp/666e041c-6192-46f9-92ee-0376c33735d6.sh: export: line 24: secrets.V5UAT_REACT_APP_COG_AUTHORISATION: bad variable name
[296](https://github.com/Halo-Solutions-Ltd/halo-web/actions/runs/3388779999/jobs/5631145881#step:10:297)
Error: Process completed with exit code 2.

Take out out and it deploys fine.

Anyone got any idea's?

Thanks in advance.

**Comment by airtower-luna** [comment]
It looks like the Actions syntaxt (`${{ ... }}`) isn't substituted before being passed from the shell. I can't say why without seeing the context, but generally it's advised to set environment variables from secrets using `env` for the step rather than injecting them into the script:

```yaml
- run: |
    # your script here
  env:
    REACT_APP_COG_AUTHORISATION: ${{ secrets.V5UAT_REACT_APP_COG_AUTH }}
```

That way the environment variable will already be set when your script runs, and you won't need to worry about escaping any special characters for the shell either.
